home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/service/impl/CooperationServiceImpl.java

```diff
@@ -0,0 +1,103 @@
+package com.softserveinc.ita.homeproject.homeservice.service.impl;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+
+import com.softserveinc.ita.homeproject.homedata.entity.Cooperation;
+import com.softserveinc.ita.homeproject.homedata.entity.House;
+import com.softserveinc.ita.homeproject.homedata.repository.CooperationRepository;
+import com.softserveinc.ita.homeproject.homedata.repository.HouseRepository;
+import com.softserveinc.ita.homeproject.homeservice.dto.CooperationDto;
+import com.softserveinc.ita.homeproject.homeservice.exception.NotFoundHomeException;
+import com.softserveinc.ita.homeproject.homeservice.mapper.ServiceMapper;
+import com.softserveinc.ita.homeproject.homeservice.service.CooperationService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class CooperationServiceImpl implements CooperationService {
+
+    private static final String NOT_FOUND_COOPERATION_FORMAT = "Can't find cooperation with given ID: %d";
+
+    private final CooperationRepository cooperationRepository;
+
+    private final ServiceMapper mapper;
+
+    private final HouseRepository houseRepository;
+
+    @Transactional
+    @Override
+    public CooperationDto createCooperation(CooperationDto createCooperationDto) {
+        Cooperation cooperation = mapper.convert(createCooperationDto, Cooperation.class);
+        cooperation.setEnabled(true);
+        cooperation.setRegisterDate(LocalDate.now());
+        cooperation.getHouses().forEach(element -> {
+            element.setCooperation(cooperation);
+            element.setCreateDate(LocalDateTime.now());
+            element.setEnabled(true);
+        });
+        cooperationRepository.save(cooperation);
+        return mapper.convert(cooperation, CooperationDto.class);
+    }
+
+    @Transactional
+    @Override
+    public CooperationDto updateCooperation(Long id, CooperationDto updateCooperationDto) {
+        Optional<Cooperation> optional = cooperationRepository.findById(id).filter(Cooperation::getEnabled);
```

Comment:
Optional have method orElseThrow(() ->new NotFoundHomeException()). With this method u can delete first "if"

Reply:
Updated!